chips/sifive/src/plic.rs

```diff
@@ -103,14 +145,14 @@ impl<const TOTAL_INTS: usize> Plic<TOTAL_INTS> {
             // at this point.
             // The priority registers indexed 0 for interrupt 1, 1 for interrupt 2, etc.
             // so we subtract 1 from the interrupt number to get the correct index.
-            priority_regs[*interrupt as usize - 1].write(priority::Priority.val(4));
+            self.registers.priority()[*interrupt as usize - 1].write(priority::Priority.val(4));
         }
         // Accept all interrupts.
-        self.registers.threshold.write(priority::Priority.val(0));
+        self.registers.threshold().write(priority::Priority.val(0));
     }
 
     pub fn disable_specific_interrupts(&self, interrupts: &[u32]) {
-        let enable_regs = &self.registers.enable[0..TOTAL_INTS.div_ceil(32)];
+        let enable_regs = self.registers.enable();
```

Comment:
```suggestion
        let enable_regs = self.registers.get_enable_regs();
```

I might suggest something like this to be clearer. A quick read of just `self.registers.enable()` in isolation reads like a method that's doing the enabling [though, the `let enable_regs = ` half does show that it's _something_ different going on]. Generally, Tock style favors clarity and longer names over conciseness.